eval/api_runner.py

```diff
@@ -302,11 +303,17 @@ def run_api_eval(args):
         # with open(prompt_file, "r") as f:
         #     prompt = f.read()
 
-        # if args.upload_url is not None:
-        #     upload_results(
-        #         results=results,
-        #         url=args.upload_url,
-        #         runner_type="api_runner",
-        #         prompt=prompt,
-        #         args=args,
-        #     )
+        if run_name is None:
+            run_name = output_file.split("/")[-1].replace(".csv", "")
+            print(
+                "Run name not provided. Using a output filename for run name:", run_name
+            )
+
+        if args.upload_url is not None:
+            upload_results(
+                results=results,
```

Comment:
it seems like `results` is only created in L289 when `logprobs` is true:
```py
results = output_df.to_dict("records")
```
And I think this might fail if `logprobs` is True and `upload_url` is provided (eg if the user is not exporting logprobs, say when using TGI or some other inference API).

Shall we update the code before the check in L285 to output `results` regardless of whether `logprobs` is provided?

Reply:
Great point, done!